python/pyspark/sql/connect/client/core.py

```diff
@@ -1614,7 +1620,10 @@ def _verify_response_integrity(
                 f"{response.session_id} != {self._session_id}"
             )
         if self._server_session_id is not None:
-            if response.server_side_session_id != self._server_session_id:
+            if (
+                response.server_side_session_id
+                and response.server_side_session_id != self._server_session_id
+            ):
```

Comment:
cc @grundprinzip FYI. Seems like `server_side_session_id` can be empty when the session does not exist, e.g. repeated `ReleaseSessionRequest` requests (see `SparkConnectReleaseSessionHandler`). This is also tested together with `test_reattach` test I added.